### PR TITLE
feat(collavre): notify when a deleted topic orphans a cron job

### PR DESCRIPTION
**Pull request by sh1nj1**
## Problem

When a Topic is deleted, any recurring cron job (`SolidQueue::RecurringTask`) that targets it via `topic_id` keeps firing but silently does nothing: `CronActionJob#perform` finds `topic_id` present, `Topic.find_by` returns `nil`, logs a warning and `return`s — no comment, no error, the task still appears in `cron_list`. The user gets **zero signal**.

This was confirmed in production: cron `cron_6245_2b2348dc` targeted `topic_id: 391`, which was deleted; the job has been a silent no-op ever since while still showing in the list.

## Fix (Decision: notify only)

Add `Collavre::Topics::OrphanedCronNotifier`, called from `TopicsController#destroy` after `topic.destroy`. It scans non-static recurring tasks, parses each task's `arguments` JSON, and for every task whose `topic_id` matches the just-deleted topic, posts a **system message** (`Comment` with `user: nil`) into that cron's creative's **Main topic**.

The recurring task itself is **left untouched** — the original cron was bound to a purpose-specific topic, so the user decides whether to re-point or cancel it after seeing the notice. System messages don't trigger AI orchestration (guarded in `comment.rb`) and do broadcast to the UI.

## Notes / scope boundary

- **No DB migration** — reuses `SolidQueue::RecurringTask` + `Comment`.
- i18n added for **en + ko** (`collavre.topics.orphaned_cron_notice`).
- Plan deviation: used the public `creative.main_topic` (the plan's `effective_origin.main_topic` is a private linked-creative resolver requiring a `Set` arg; the test asserts against `creative.main_topic`).
- **Fires on future deletions only.** Already-orphaned crons (e.g. the production `cron_6245` → topic 391) are not retro-notified; surfacing those is a separate one-off.

## Tests

- New `OrphanedCronNotifierTest` (4 cases): posts to main topic + includes cron key/topic name; leaves the task in place; no-op for a different topic; no-op for `topic_id: nil` (main-topic crons).
- Full `topics_controller_test` + notifier test: **30 runs, 0 failures**. Rubocop clean.

Edge cases: cron targeting a different topic (no-op), `topic_id: nil` (no-op), creative not found (skip), Main topic can't be deleted so `main_topic` always exists.

### CODE REVIEW

**Comment by sh1nj1** [comment]
## 코드 리뷰 (Corex) — 변경 요청 2건

전반적으로 설계(옵션1: 알림만, cron 유지)는 깔끔하고 테스트도 통과(9 runs, 0 failures)합니다. 다만 머지 전 2건 수정이 필요합니다.

### 1. 중복 코드 (must-fix)
`OrphanedCronNotifier#parse_arguments` 와 `#parse_creative_id_from_key` 가 `Collavre::Tools::CronListService` 의 동일 메서드와 **완전히 동일**합니다 (문자 단위로 일치). RecurringTask args 파싱 로직은 한 곳에만 있어야 합니다.
- 제안: 두 메서드를 공유 concern/module (예: `Collavre::Crons::ArgumentParsing`)로 추출해 `CronListService` 와 `OrphanedCronNotifier` 가 함께 include.

### 2. 삭제 후 best-effort 알림이 핵심 삭제를 깨뜨릴 수 있음 (correctness)
`topics_controller#destroy`:
```ruby
topic.destroy                       # 비가역
OrphanedCronNotifier.new(...).call  # 여기서 raise 하면?
broadcast_topic_event("deleted", ...)# 실행 안 됨 → UI stale
head :no_content                    # 대신 500 반환
```
`Comment.create!` 등이 예외를 던지면 토픽은 이미 삭제됐는데 액션은 500 을 반환하고, `broadcast_topic_event("deleted")` 도 건너뜁니다(클라이언트 UI 가 삭제 반영 못 함). 알림은 부가 기능이므로 핵심 삭제 흐름을 절대 깨면 안 됩니다.
- 제안: notifier 호출을 `rescue => e` 로 감싸 `Rails.logger.error` 로 남기고 삼키기 (broadcast/head 는 항상 실행되도록).

보안/디자인토큰/죽은코드: 특이사항 없음. (message 가 시스템 코멘트에 보간되나 사용자 본인 크론 메시지라 위험 낮음.)

File: engines/collavre/app/services/collavre/topics/orphaned_cron_notifier.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        creative = Creative.find_by(id: creative_id)
+        return unless creative
+
+        main_topic = creative.main_topic
+        return unless main_topic
+
+        Comment.create!(
+          creative: creative,
+          topic_id: main_topic.id,
+          user: nil, # System message
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre/app/services/collavre/topics/orphaned_cron_notifier.rb:56]: Fixed in e800a6ec. Confirmed real: `assign_default_user` does `self.user ||= Collavre.current_user`, and in `TopicsController#destroy` the current user is the admin deleting the topic — so `user: nil` was being overwritten and `dispatch_to_orchestration` would have fired (admin is non-AI, creative present, not inbox). Now sets `skip_default_user: true` on the notice so it stays a system message and never dispatches. Added a regression test that sets `Collavre::Current.user` and asserts the notice keeps `user_id` nil.

**Comment by sh1nj1** [comment]
## 재리뷰 — 변경 요청 3건 모두 반영 완료 (e800a6ec)

| # | 이슈 | 처리 |
|---|------|------|
| 1 | **중복 코드** | `parse_arguments` / `parse_creative_id_from_key` 를 `Collavre::Crons::RecurringTaskArguments` 모듈로 추출. `CronListService` 와 `OrphanedCronNotifier` 둘 다 include — 파싱 로직 단일 소스화. |
| 2 | **삭제 흐름 보호** | `TopicsController#destroy` 의 notifier 호출을 `begin/rescue StandardError` 로 감쌈. 알림 실패 시 `Rails.logger.error` 로 남기고 삼켜 → `broadcast_topic_event("deleted")` 와 `head :no_content` 는 항상 실행. |
| 3 | **Codex P1 (system message 귀속)** | 확인 결과 실제 결함: `assign_default_user` 가 `self.user ||= Collavre.current_user` 이고 destroy 시 current_user 는 삭제 admin → `user: nil` 이 덮어써지고 `dispatch_to_orchestration` 발화 위험. `skip_default_user: true` 추가로 시스템 메시지 유지 + AI 미발화. 회귀 테스트(`Collavre::Current.user` 세팅 후 `user_id` nil 단언) 추가. |

**검증:** notifier 5 runs / topics_controller 26 runs / cron_list 5 runs — 0 failures. Rubocop clean. pre-push 훅(rubocop + 전체 테스트) 통과.

머지는 사용자 승인 대기 중입니다.

**Comment by sh1nj1** [comment]
## ✅ 재리뷰 완료 — 변경요청 3건 모두 반영 (`e800a6ec`)

| # | 이슈 | 상태 |
|---|------|------|
| 1 | **중복코드** — RecurringTask args 파싱 2곳 중복 | ✅ `Collavre::Crons::RecurringTaskArguments` 모듈로 추출, `CronListService`·`OrphanedCronNotifier` 둘 다 include. 복사본 제거 |
| 2 | **best-effort 알림이 핵심 삭제 깨뜨림** | ✅ `destroy`에서 notifier를 `rescue StandardError` + `Rails.logger.error`로 감쌈. `broadcast_topic_event('deleted')`·`head`는 항상 실행 |
| 3 | **Codex P1 — 시스템 메시지 오귀속** | ✅ notice에 `skip_default_user: true` 설정. `assign_default_user`(comment.rb:154)·`dispatch_to_orchestration`(comment.rb:137) 둘 다 가드되어 `user: nil` 유지 + AI 오케스트레이션 미발화. 회귀 테스트 추가 |

**검증:** notifier 5 runs(P1 회귀 포함) + cron_list 5 + topics_controller 26 → 0 failures. Rubocop clean. pre-push 전체 스위트 통과.

보안·디자인토큰·죽은코드: 특이사항 없음. **머지 승인 대기** (룰: 머지는 사용자 승인 필요).

File: engines/collavre/app/services/collavre/topics/orphaned_cron_notifier.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+          [ task, args ]
+        end
+      end
+
+      def notify_for(task, args)
+        creative_id = args["creative_id"] || parse_creative_id_from_key(task.key)
+        creative = Creative.find_by(id: creative_id)
+        return unless creative
+
+        main_topic = creative.main_topic
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre/app/services/collavre/topics/orphaned_cron_notifier.rb]: Fixed in 00f77760. Confirmed real: `CronCreateService#resolve_topic_id` resolves the topic on `creative.effective_origin` but the recurring task stores the original (linked/child) `creative_id`. `Comment#use_origin_creative` (before_validation) then rewrites the notice to the origin while `assign_main_topic` early-returns (topic_id already set) — so `creative.main_topic` left the notice pinned to the linked creative's Main topic, mismatched against the origin and hidden from normal navigation. Now uses `creative.effective_origin.main_topic`; no-op for non-linked creatives (`effective_origin` returns self). Added a regression test for the linked-creative path.

**Comment by sh1nj1** [comment]
## ✅ Codex P2 반영 (`00f77760`)

**linked/child 크리에이티브 cron의 알림이 엉뚱한 토픽에 게시되던 문제 수정.**

확인 결과 실제 결함이었습니다:
- `CronCreateService#resolve_topic_id`는 `creative.effective_origin`에서 토픽을 해석하지만, RecurringTask에는 **원본(linked/child) creative_id**를 저장.
- `Comment#use_origin_creative`(before_validation)가 알림 코멘트의 creative를 origin으로 재작성 → 하지만 `assign_main_topic`은 topic_id가 이미 세팅돼 early-return.
- 결과적으로 `creative.main_topic`은 **linked 크리에이티브의 Main 토픽**에 알림을 고정 → origin 크리에이티브와 불일치, 일반 토픽 네비게이션에서 숨김.

**수정:** `creative.effective_origin.main_topic` 사용. non-linked 크리에이티브는 `effective_origin`이 self를 반환하므로 동작 변화 없음(no-op). linked-creative 경로 회귀 테스트 추가.

**검증:** notifier 6 runs(linked-creative 회귀 포함) → 0 failures. Rubocop clean. pre-push 전체 스위트 통과.

> 참고: 앞선 리뷰에서 plan의 `effective_origin.main_topic`를 `creative.main_topic`으로 바꾼 판단이 틀렸습니다 — `effective_origin`은 public이고 기본 인자(`Set.new`)가 있어 인자 없이 호출 가능합니다(`cron_create_service`·`comment`도 동일하게 사용). Codex 지적이 옳았고, 원래 plan 의도를 복원합니다.

머지 승인 대기 중입니다.
